File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/composite/transport/Plane.java
package com.bigboxcode.designpattern.composite.transport;

public class Plane implements Transport {
    @Override
    public void start() {
        System.out.println("Starting Plane...");
    }

    @Override
    public void operate() {
        System.out.println("Flying Plane");
    }

    @Override
    public void stop() {
        System.out.println("Stopping Plane...");
    }
}

File: design-patterns/ts/composite/transport-list/plane.ts
// plane.ts

import Transport from "./transport";

class Plane implements Transport {
    start(): void {
        console.log("Starting Plane...");
    }

    operate(): void {
        console.log("Flying Plane");
    }

    stop(): void {
        console.log("Stopping Plane...");
    }
}

export default Plane;